execution_engine_testing/tests/src/test/system_contracts/mint.rs

```diff
@@ -0,0 +1,206 @@
+use casper_engine_test_support::{
+    auction, ExecuteRequestBuilder, LmdbWasmTestBuilder, DEFAULT_ACCOUNT_ADDR,
+};
+use casper_types::{runtime_args, RuntimeArgs, URef, U512};
+
+use tempfile::TempDir;
+
+const TEST_DELEGATOR_INITIAL_ACCOUNT_BALANCE: u64 = 1_000_000 * 1_000_000_000;
+
+const CONTRACT_BURN: &str = "burn.wasm";
+const CONTRACT_TRANSFER_TO_NAMED_PURSE: &str = "transfer_to_named_purse.wasm";
+
+const ARG_AMOUNT: &str = "amount";
+
+const ARG_PURSE_NAME: &str = "purse_name";
+
+#[ignore]
+#[test]
+fn should_burn_tokens_from_provided_purse() {
```

Comment:
```suggestion
fn should_empty_purse_when_burning_above_balance() {
```
or something like that